packages/flutter/lib/src/material/autocomplete.dart

```diff
@@ -173,55 +173,119 @@ class _AutocompleteOptions<T extends Object> extends StatelessWidget {
-          child: ListView.builder(
-            padding: EdgeInsets.zero,
-            shrinkWrap: true,
-            itemCount: options.length,
-            itemBuilder: (BuildContext context, int index) {
-              final T option = options.elementAt(index);
-              return InkWell(
-                onTap: () {
-                  onSelected(option);
-                },
-                child: Builder(
-                  builder: (BuildContext context) {
-                    final bool highlight = AutocompleteHighlightedOption.of(context) == index;
-                    if (highlight) {
-                      SchedulerBinding.instance.addPostFrameCallback((Duration timeStamp) {
-                        Scrollable.ensureVisible(context, alignment: 0.5);
-                      }, debugLabel: 'AutocompleteOptions.ensureVisible');
-                    }
-                    return Container(
-                      color: highlight ? Theme.of(context).focusColor : null,
-                      padding: const EdgeInsets.all(16.0),
-                      child: Text(displayStringForOption(option)),
-                    );
-                  }
-                ),
-              );
-            },
+          child: _AutocompleteOptionsList<T>(
+            displayStringForOption: displayStringForOption,
+            highlightedIndex: highlightedIndex,
+            onSelected: onSelected,
+            options: options,
           ),
         ),
       ),
     );
   }
 }
+
+class _AutocompleteOptionsList<T extends Object> extends StatefulWidget {
+  const _AutocompleteOptionsList({
+    required this.displayStringForOption,
+    required this.highlightedIndex,
+    required this.onSelected,
+    required this.options,
+  });
+
+  final AutocompleteOptionToString<T> displayStringForOption;
+  final int highlightedIndex;
+  final AutocompleteOnSelected<T> onSelected;
+  final Iterable<T> options;
+
+  @override
+  State<_AutocompleteOptionsList<T>> createState() => _AutocompleteOptionsListState<T>();
+}
+
+class _AutocompleteOptionsListState<T extends Object> extends State<_AutocompleteOptionsList<T>> {
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void didUpdateWidget(_AutocompleteOptionsList<T> oldWidget) {
+    // Call super first.
+    super.didUpdateWidget(oldWidget);
+
+    if (widget.highlightedIndex != oldWidget.highlightedIndex) {
+      SchedulerBinding.instance.addPostFrameCallback((Duration timeStamp) {
+        if (!mounted) {
+          return;
+        }
+        final BuildContext? highlightedContext =
+            GlobalObjectKey(widget.options.elementAt(widget.highlightedIndex)).currentContext;
+        if (highlightedContext == null) {
+          _scrollController.jumpTo(
+            widget.highlightedIndex == 0 ? 0.0 : _scrollController.position.maxScrollExtent,
+          );
+        } else {
+          Scrollable.ensureVisible(
+            highlightedContext,
+            alignment: 0.5,
```

Comment:
This existed before this PR, but its unclear to me what this value represents.

Reply:
It puts the item in the middle of the scrollable, which seems to work nicely for Autocomplete. I thought about commenting this here, but I think it's best to leave describing what it does up to Scrollable.ensureVisible.